code/modules/modular_computers/networking/machinery/telecomms.dm

```diff
@@ -0,0 +1,328 @@
+	// cell, assuming we have a fuctioning comms maser and the overmap site has a telecomms hub.
+	LAZYINITLIST(levels)
+	levels |= GetConnectedZlevels(z)
+	var/obj/effect/overmap/O = global.overmap_sectors["[z]"]
+	if(istype(O) && length(O.comms_masers))
+		for(var/obj/machinery/shipcomms/broadcaster/our_maser in O.comms_masers)
+			levels |= our_maser.get_available_z_levels()
+
+	// Find networks in our z-chunk that have a hub with a channel with the same name, security and frequency.
+	var/list/receiving_networks = list(network)
+	var/datum/radio_channel/channel = get_channel("[frequency]")
+	for(var/obj/machinery/network/telecomms_hub/other_hub in global.telecomms_hubs)
+		if(!global_radio_broadcaster && !(other_hub.z in levels))
+			continue
+		var/datum/radio_channel/other_channel = other_hub.get_channel("[frequency]")
+		if(!other_channel || other_channel.name != channel.name)
+			continue
+		if(islist(other_channel.secured) && islist(channel.secured))
+			for(var/access_req in other_channel.secured)
+				if(!(access_req in channel.secured))
+					continue
+			for(var/access_req in channel.secured)
+				if(!(access_req in other_channel.secured))
+					continue
+		var/datum/extension/network_device/other_network_device = get_extension(other_hub, /datum/extension/network_device)
+		var/datum/computer_network/other_network = other_network_device?.get_network()
+		if(other_network)
+			receiving_networks |= other_network
+
+	// Broadcast to all radio devices in the receiving networks.
+	for(var/datum/computer_network/use_network in receiving_networks)
+		for(var/weakref/W as anything in use_network.connected_radios)
+			var/obj/item/radio/R = W.resolve()
+			if(!istype(R) || QDELETED(R) || !R.can_receive_message(use_network))
+				continue
+			var/turf/speaking_from = get_turf(R)
+			if(!speaking_from)
+				continue
+			if(!channel || !R.can_decrypt(channel.secured))
+				continue
+			var/list/check_channels = R.get_available_channels()
+			if(!LAZYACCESS(check_channels, channel))
+				continue
+			var/turf/T = get_turf(R)
+			var/overmap_object = istype(T) && global.overmap_sectors["[T.z]"]
+			for(var/mob/M in hearers(R.canhear_range, speaking_from))
+				LAZYSET(., M, overmap_object)
+
+/obj/machinery/network/telecomms_hub/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+
+	var/list/data = list()
+	var/datum/extension/network_device/device = get_extension(src, /datum/extension/network_device)
+	data["network_id"] = device.network_tag
+	data["authenticated"] = (access_keycard_auth in user.GetAccess())
+	var/list/data_channels = list()
+	for(var/datum/radio_channel/channel_datum in channels)
+		var/list/channel_data = list()
+		channel_data["channel_name"] =   channel_datum.name
+		channel_data["channel_freq"] =   channel_datum.frequency
+		channel_data["channel_key"] =    channel_datum.key
+		channel_data["channel_colour"] = channel_datum.color || COLOR_BLACK
```

Comment:
Not important but would prefer if it was all color instead of switching between color and colour. Color is the more common variant in the repo. If you end up doing so make sure to update your .tmpl too!